src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,39 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+// Parent Class of Address, Email and Phone
+public class Contact {
+
+        private boolean isPrivate;
+        public String value;
+        
+        public Contact(String contact, boolean isPrivate) throws IllegalValueException {
+            this.isPrivate = isPrivate;
+            String trimmedContact = contact.trim();
+            this.value = trimmedContact;
+        }
+        
+        
+        //Override
+        public boolean equals(Object otherContact) {
```

Comment:
Note that this can be "equal" if the class we are comparing are Email class vs Address class, and their values are the same. Comparison should be done in the child classes for more detailed checks.